corehq/apps/app_manager/xform.py

```diff
@@ -976,20 +979,37 @@ def _get_label_ref(self, prompt):
         return None
 
     def resolve_path(self, path, path_context=""):
-        if path == "":
-            return path_context
-        elif path is None:
+        '''
+            input: path with type ConditionalCaseUpdate
+            output: typen str
+        '''
+        try:
+            path_str = path.question_path
+        except AttributeError:
+            path_str = path
+        try:
+            path_context_str = path_context.question_path
+        except AttributeError:
+            path_context_str = path_context
+        if path_str == "":
+            return path_context_str
+        elif path_str is None:
             raise CaseError("Every case must have a name")
-        elif path[0] == "/":
-            return path
+        elif path_str[0] == "/":
+            return path_str
         elif not path_context:
-            return "/%s/%s" % (self.data_node.tag_name, path)
+            return "/%s/%s" % (self.data_node.tag_name, path_str)
         else:
-            return "%s/%s" % (path_context, path)
+            return "%s/%s" % (path_context_str, path_str)
 
     def hashtag_path(self, path):
-        for hashtag, replaces in hashtag_replacements:
-            path = re.sub(replaces, hashtag, path)
+        from corehq.apps.app_manager.models import ConditionalCaseUpdate
```

Comment:
Nitpick similar to above comment: you could start this function with `path = getattr(path, "question_path", path)` instead of the if/else where both branches have the same loop.

Reply:
ah thank you I was trying to remember a better way to handle this logic-- this is great